starfish/imagestack/indexing_utils.py

```diff
@@ -26,8 +27,42 @@ def convert_to_selector(
     return return_dict
 
 
-def index_keep_dimensions(
-        data: xr.DataArray, indexers: Mapping[str, Union[int, slice]]) -> xr.DataArray:
+def convert_coords_to_indices(array: xr.DataArray,
+                              indexers: Mapping[Coordinates, Union[Number, Tuple[Number, Number]]]
+                              ) -> Dict[Axes, Union[int, tuple]]:
+    """
+    Convert mapping of physical coordinates to value or range to mapping of corresponding Axes and
+    positional coordinates.
+
+    Parameters
+    ----------
+    array: xr.DataArray
+        The xarray with both physical and positional coordinates.
+    indexers: Mapping[Coordinates, Union[int, tuple]]
+        Mapping of physical coordinates to value or range
+
+    Returns
+    -------
+    Mapping[Axes, Union[int, tuple]]:
+        Mapping of Axes and positional indices that correspond to the given physical indices.
+
+    """
+    axes_indexers: Dict[Axes, Union[int, tuple]] = dict()
+    if Coordinates.X in indexers:
+        idx_x = find_nearest(array[Coordinates.X.value], indexers[Coordinates.X])
+        axes_indexers[Axes.X] = idx_x
+    if Coordinates.Y in indexers:
+        idx_y = find_nearest(array[Coordinates.Y.value], indexers[Coordinates.Y])
+        axes_indexers[Axes.Y] = idx_y
+    if Coordinates.Z in indexers:
```

Comment:
What if someone specified Coordinates.Z in indexers, but the image has only one zplane?  I would assume it would bomb out with a nasty error since the array would not have z coordinates.

Reply:
Changed to use isel for z case!